vault/identity_store_util.go

```diff
@@ -341,14 +341,15 @@ func (i *IdentityStore) upsertEntityInTxn(ctx context.Context, txn *memdb.Txn, e
 			case intErr != nil:
 				return intErr
 			}
+
```

Comment:
After the merge it would return nil (or an error). Since you've removed the `default` case, instead execution will continue on down the remainder of this function. Is this what we want?

Reply:
Yes. This is what we want. The return criteria will be met only when we run mergeEntity. That's why the `default:` was doing a return. Now, we want to do that only when the case condition is satisfied. So, I think this is fine. Thanks for asking though. It's a neat observation.